module01/ex05/ex05.md

```diff
@@ -1,37 +1,84 @@
-# Exercise 05 - Working with lists
+# Exercise 05 - Bank Account
 
-|                         |                                 |
-| -----------------------:| ------------------------------- |
-|   Turn-in directory:    |  ex05                           |
-|   Files to turn in:     |  eval.py                        |
-|   Forbidden functions   |  while                          |
-|   Remarks:              |  use zip & enumerate            |
+|                         |                     |
+| -----------------------:| ------------------- |
+|   Turn-in directory:    |  ex05/              |
+|   Files to turn in:     |  the_bank.py        |
+|   Forbidden functions:  |  None               |
+|   Remarks:              |  n/a                |
 
 ## Objective:
-The goal of the exercise is to discover 2 useful methods for listsm tuplesm dictionnaries (iterable class
-objects more generally) named `zip` and `enumerate`.
+The goals of this excercise is to discover new built-in functions, deepen the class manipulation and
+to be aware of the posibility to modify istanced objects.
+In this excercise you learn how to modify or add attributes to an object.
 
 
-## Instructions:
-Code a class `Evaluator`, that has two static functions named `zip_evaluate` and `enumerate_evaluate`.
+# Instructions:
+It's all about security. Have a look at the class named `Account` in the snippet of code below.
 
-The goal of these 2 functions is to compute the sum of the lengths of every `words` of a given list weighted by a list a `coefs`.
+```py
+# in the_bank.py
+class Account(object):
+
+    ID_COUNT = 1
 
-The lists `coefs` and `words` have to be the same length. If this is not the case, the function should return -1.
+    def __init__(self, name, **kwargs):
+        self.id = self.ID_COUNT
+        self.name = name
+        self.__dict__.update(kwargs)
+        if hasattr(self, 'value'):
+            self.value = 0
+        Account.ID_COUNT += 1
+    
+    def transfer(self, amount):
+        self.value += amount
+
+```
 
-You have to obtain the desired result using `zip` in the `zip_evaluate` function, and with `enumerate` in the `enumerate_evaluate` function.
+Now, it is your turn to code a class named `Bank`!  
+Its purpose will be to handle the security part of each transfer attempt.
 
+Security means checking if the `Account` is:
+* the right object,
+* not corrupted,
+* and stores enough money to complete the transfer.
+
+How do we define if a bank account is corrupted? A corrupted bank account has:
+* an even number of attributes,
+* an attribute starting with `b`,
+* no attribute starting with `zip` or `addr`,
+* no attribute `name`, `id` and `value`.
+
+A transaction is invalid if `amount < 0` or if the amount is larger than the available funds of the sending account.
```

Comment:
available funds -> balance ?

Reply:
correction copy on the subject for the holygraph

Reply:
fix